app/javascript/packs/inertia.js

```diff
@@ -14,6 +14,13 @@ router.on("before", (event) => {
       "X-CSRF-Token": token,
     };
   }
+
+  // Track previous route for navigation
+  const currentUrl = window.location.pathname;
+  const newUrl = typeof event.detail.visit.url === "string" ? event.detail.visit.url : event.detail.visit.url.pathname;
+  if (currentUrl !== newUrl) {
+    sessionStorage.setItem("inertia_previous_route", currentUrl);
+  }
```

Comment:
Added an event handler in Inertia's before hook to track the previous route in sessionStorage before navigation.